Products/ZenUI3/browser/resources/js/zenoss/devdetail.js

```diff
@@ -436,28 +436,33 @@ var dev_admin = Ext.create('Zenoss.devicemanagement.Administration', {
 
 // find out how many columns the graph panel should be based on
 // on the width of the detail panel
-var center_panel_width = Ext.getCmp('center_panel').getEl().getWidth() - 275,
-    extra_column_threshold = 1000;
+var center_panel_width = Ext.getCmp('center_panel').getEl().getWidth() - 275;
+var extra_column_threshold = 1000;
+var userColumns = Zenoss.settings.graphColumns;
+// graphColumns 0 is "Auto"
+userColumns = ((Zenoss.settings.graphColumns === 0) && (center_panel_width > extra_column_threshold)) ? 2 : 1;
```

Comment:
does this work if number of columns is set to 3?

Reply:
absolutely.

Reply:
Line +441 sets userColumns to the settings value

Reply:
im parsing js with my eyeballs, but im pretty sure line 443 will always evaluate, and always return either 2 or 1.

Reply:
only when `Zenoss.settings.graphColumns === 0`  that's the "Auto" check